pa-rserve/src/main/java/org/ow2/parserve/RexpConvert.java

```diff
@@ -0,0 +1,443 @@
+package org.ow2.parserve;
+
+import org.ow2.parengine.RObject;
+import org.rosuda.REngine.*;
+
+import java.beans.BeanInfo;
+import java.beans.Introspector;
+import java.beans.PropertyDescriptor;
+import java.lang.reflect.Array;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.util.*;
+
+/**
+ * Routines to convert between Java Objects and R expressions.
+ * <p/>
+ * <table>
+ * <tr><td> null                                                <td> REXPNull
+ * <tr><td> boolean, Boolean, boolean[], Boolean[]              <td> REXPLogical
+ * <tr><td> int, Integer, int[], Integer[]                      <td> REXPInteger
+ * <tr><td> double, Double, double[], double[][], Double[]	<td> REXPDouble
+ * <tr><td> String, String[]					<td> REXPString
+ * <tr><td> byte[]						<td> REXPRaw
+ * <tr><td> Enum						<td> REXPString
+ * <tr><td> Object[], List, Map					<td> REXPGenericVector
+ * <tr><td> RObject, java bean (experimental)                   <td> REXPGenericVector
+ * </table>
+ *
+ * @author Martyn Tebby
```

Comment:
Would be nice to have a reference to the repository or resource from where it has been retrieved. It could be useful if a bug is discovered.